src/features/player/index.js

```diff
@@ -50,13 +58,87 @@ class Player extends Component {
             let currentTick = 0;
             const ticksPerFrame = 5;
 
+            const {
+                hairColour,
+                skinColour,
+                armourColour,
+                clothesColour,
+            } = this.props.dialog.appearance;
+
             const draw = frame => {
                 // don't allow invalid frames
                 if (frame > 7 || frame < 0) frame = 0;
 
                 ctx.clearRect(0, 0, SPRITE_SIZE, SPRITE_SIZE);
+                // Draw player hair
+                ctx.filter = `hue-rotate(${hairColour - 10}deg)`; //createFilterFromHexCode(hairColour);
+                ctx.drawImage(
+                    this.sprite.hair,
+                    frame * SPRITE_SIZE,
+                    spriteLine,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE,
+                    0,
+                    0,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE
+                );
+                // Draw player eyes
+                ctx.filter = 'none';
+                ctx.drawImage(
+                    this.sprite.eyes,
+                    frame * SPRITE_SIZE,
+                    spriteLine,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE,
+                    0,
+                    0,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE
+                );
+                // Draw player skin
+                ctx.filter = `hue-rotate(${skinColour - 10}deg)`; //createFilterFromHexCode(skinColour);
+                ctx.drawImage(
+                    this.sprite.skin,
+                    frame * SPRITE_SIZE,
+                    spriteLine,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE,
+                    0,
+                    0,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE
+                );
+                // Draw player armour
+                ctx.filter = `hue-rotate(${armourColour - 10}deg)`; //createFilterFromHexCode(armourColour);
+                ctx.drawImage(
+                    this.sprite.armour,
+                    frame * SPRITE_SIZE,
+                    spriteLine,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE,
+                    0,
+                    0,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE
+                );
+                // Draw player clothes
+                ctx.filter = `hue-rotate(${clothesColour - 10}deg)`; //createFilterFromHexCode(clothesColour);
+                ctx.drawImage(
+                    this.sprite.clothes,
+                    frame * SPRITE_SIZE,
+                    spriteLine,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE,
+                    0,
+                    0,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE
+                );
+                // Draw player outline
+                ctx.filter = 'none';
```

Comment:
Would it be possible to turn all of these (and possibly the one below them) into a loop?

Reply:
Previous attempts of mine to iterate over `this.sprite` weren't successful. Do you mind suggesting a way to loop over it?

Reply:
Does `for (var key in this.sprite)` not work? (You'd have to special case the 'onload' key). But if it doesn't work easily, then it's not a big deal.